### ISSUE TITLE
NullPointerException with a non initialised variable in a Zest script

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#2871]
Steps to reproduce the issue:
1. Run ZAP with Zest add-on;
2. Run the following stand alone Zest script:

``` JSON
{
  "about": "This is a Zest script. For more details about Zest visit https://developer.mozilla.org/en-US/docs/Zest",
  "zestVersion": "0.3",
  "title": "X",
  "description": "A template standalone script",
  "prefix": "",
  "type": "StandAlone",
  "parameters": {
    "tokenStart": "{{",
    "tokenEnd": "}}",
    "tokens": {},
    "elementType": "ZestVariables"
  },
  "statements": [
    {
      "url": "http://example.com/",
      "data": "",
      "method": "GET",
      "headers": "",
      "assertions": [],
      "followRedirects": false,
      "cookies": [],
      "index": 1,
      "enabled": true,
      "elementType": "ZestRequest"
    },
    {
      "fieldDefinition": {
        "formIndex": 3,
        "fieldName": "CSRFToken",
        "elementType": "ZestFieldDefinition"
      },
      "variableName": "csrf1",
      "index": 2,
      "enabled": true,
      "elementType": "ZestAssignFieldValue"
    },
    {
      "url": "http://example.com/login",
      "data": "CSRFToken\u003d{{csrf1}}",
      "method": "POST",
      "headers": "Content-Type: application/x-www-form-urlencoded",
      "assertions": [],
      "followRedirects": false,
      "cookies": [],
      "index": 3,
      "enabled": true,
      "elementType": "ZestRequest"
    }
  ],
  "authentication": [],
  "index": 0,
  "enabled": true,
  "elementType": "ZestScript"
}
```

3.Note that in the output panel it shows:

```
java.lang.NullPointerExceptionjava.lang.NullPointerException
```

Remarks:
Using the Zest library with debug info it shows the following stack trace:

```
71584 [ZAP-ScriptExecutor-X2] ERROR org.zaproxy.zap.extension.scripts.OutputPanel  - 
java.lang.NullPointerException
    at java.lang.String.replace(String.java:2240)
    at org.mozilla.zest.core.v1.ZestVariables.replaceInString(ZestVariables.java:200)
    at org.mozilla.zest.core.v1.ZestVariables.replaceInString(ZestVariables.java:230)
    at org.mozilla.zest.core.v1.ZestRequest.replaceTokens(ZestRequest.java:305)
    at org.mozilla.zest.impl.ZestBasicRunner.runStatement(ZestBasicRunner.java:169)
    at org.zaproxy.zap.extension.zest.ZestZapRunner.runStatement(Unknown Source)
    at org.mozilla.zest.impl.ZestBasicRunner.run(ZestBasicRunner.java:142)
    at org.zaproxy.zap.extension.zest.ZestZapRunner.run(Unknown Source)
    at org.mozilla.zest.impl.ZestBasicRunner.run(ZestBasicRunner.java:98)
    at org.zaproxy.zap.extension.zest.ZestZapRunner.run(Unknown Source)
    at org.mozilla.zest.impl.ZestBasicRunner.runScript(ZestBasicRunner.java:435)
    at org.mozilla.zest.impl.ZestScriptEngine.eval(ZestScriptEngine.java:49)
    at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
    at org.zaproxy.zap.extension.script.ExtensionScript.invokeScriptImpl(ExtensionScript.java:1037)
    at org.zaproxy.zap.extension.script.ExtensionScript.invokeScript(ExtensionScript.java:976)
    at org.zaproxy.zap.extension.scripts.ConsolePanel$ScriptExecutorThread.run(Unknown Source)
```

For reference:
https://github.com/mozilla/zest/blob/a8d1f40d0d80e25e5f12830fb657ed05244694e1/src/org/mozilla/zest/core/v1/ZestVariables.java#L200

ZAP version:
Version 2.5.0

Add-on:
"Zest - Graphical Security Scripting Language" version 22

**Comment by thc202** [comment]
This was fixed in the Zest library mozilla/zest#80 (which needs to be updated in Zest add-on once the new version is available).

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#985.
